### PR TITLE
Support edge case with tests which have no difference except being in different assemblies

### PR DESCRIPTION
**Pull request by van800**
Fix RIDER-24636 Rider runs wrong tests if two projects has the same namespace name

### CODE REVIEW

File: unity/EditorPlugin/AfterUnity56/UnitTesting/TestEvent.cs [hunk 1]
```diff
@@ -23,14 +23,16 @@ public TestEvent(EventType type, TestInternalEvent tEvent)
   public class TestInternalEvent
   {
     public string myID;
+    public readonly string myAssemblyName;
```
Comment (citizenmatt) [inline unity/EditorPlugin/AfterUnity56/UnitTesting/TestEvent.cs]: Make the other field `readonly` too?

Comment (van800) [reply unity/EditorPlugin/AfterUnity56/UnitTesting/TestEvent.cs]: Well spotted! I have to actually remove readonly here. Otherwise it will not get restored after AppDomain reload.
